### ISSUE TITLE
Allowing `aria-hidden` on visible content excludes low vision uses

### ISSUE THREAD

**Issue by steverep** [w3c/aria#583]
The `aria-hidden` specification currently includes the following cautionary use case:

> Authors may, with caution, use aria-hidden to hide visibly rendered content from assistive technologies only if the act of hiding this content is intended to improve the experience for users of assistive technologies by removing redundant or extraneous content. Authors using aria-hidden to hide visible content from screen readers must ensure that identical or equivalent meaning and functionality is exposed to assistive technologies.

I would like to understand where there is a valid use case for doing this, that would not be much more appropriately handled using `aria-label=""` instead.  Using `aria-label` in this way is a better approach for the following reasons:

1. It is consistent with HTML specifications to use `alt=""` on `<img>` for decorative images
2.  It is consistent with CSS specifications to use `content: ... / ""` for decorative uses of `::before` or `::after`.
3. Visual focus indicators are not advised on how to deal with content that is visible but supposedly hidden.  Inconsistent behaviors already arise here based on my various tests with different user agents and AT.  When `aria-label=""` is used instead, the behavior is predictably synonymous with `<img alt="" ...>`.  This is important for low vision users of screen readers.
4. No guidance is given on what, if any, accessible name should be presented to AT on mouse hover (a mainly low vision user method which all screen readers support).  Currently, most tests of browser and screen reader combinations indicate that an accessible name computation still tries to take place in this case, and I'd argue this is good practice for low vision users.  For non-text content such as font icons or other CSS inserted content, the name often reverts to something generic or perhaps the surrounding anchor's URL if present.  Again, however, behavior using `aria-label=""` is already well in line with `alt=""`.

Please see the [thread from  WCAG 2.1 under discussion for greater detail](https://github.com/w3c/wcag21/issues/297#issuecomment-303139969).

So, if I am correct that there are no real use cases where it is appropriate to walk this cautionary path, I propose it be removed and clarification be given to `aria-label` on decorative content.  If I'm wrong, I'd like to understand any such cases in detail to ensure all low vision needs are met.  Thanks.

**Comment by steverep** [comment]
A quick follow-up is that there is a current implication in rule step 2C of the accessible name computation to essentially ignore `aria-label=""`:

> Otherwise, if computing a name, and if the current node has a non-empty aria-label attribute:

I'd strongly suggest changing this to omit "non-empty".

**Comment by eps1lon** [comment]
We use it for a breadcrumb component. 

```html
<nav aria-label="breadcrumbs">
  <ol>
    <li>first</li>
    <li aria-hidden="true"> / </li>
    <li>second</li>
  </ol>
</nav>
```

Using CSS limits customizability of this component and makes the separators not user-selectable.
Including the separator in the item might be more confusing. Announcing it to AT by using different markup might cause other issues.

**Comment by ZoeBijl** [comment]
@eps1lon as it stands the APG advises to add the separators with CSS. Can you explain to me why you want them to be user selectable?

**Comment by eps1lon** [comment]
>  Can you explain to me why you want them to be user selectable?

I personally prefer text content to be selectable. Displaying text and not being able to copy it is frustrating. You wouldn't exclude commas from text-selection either.

**Comment by ZoeBijl** [comment]
> I personally prefer text content to be selectable. Displaying text and not being able to copy it is frustrating. You wouldn't exclude commas from text-selection either.

A visual separator isn’t the same as punctuation. The bullets or numbers in front of either an unordered or ordered list aren’t selectable. If you visualise a breadcrumb as being a horizontal ordered list you probably end up at the same conclusion.

**Comment by eps1lon** [comment]
> A visual separator isn’t the same as punctuation.

It wasn't meant to be taken as an equivalent.

> The bullets or numbers in front of either an unordered or ordered list aren’t selectable. 

Which is why I never use them. The visual separator includes a newline which is user-selectable though.

Anyway I prefer displayed text content to be user selectable. That is my use case.

**Comment by ZoeBijl** [comment]
Your breadcrumb example doesn’t align with the ARIA WGs advice or best practices. You can read our guidance and view our code example in the ARIA Authoring Practices’ [breadcrumb design pattern](http://w3c.github.io/aria-practices/#breadcrumb).

**Comment by jnurthen** [comment]
Going back to the original point of this issue `aria-label=""` is a no-op in the spec. 

At this point we are not going to change behaviour to make `aria-label=""` override content. If aria-label is empty the name calculation will move on to the next step
